install.rst

```diff
@@ -30,17 +33,82 @@ If you use **Nix**, then you can install PostgREST from nixpkgs.
 
   nix-env -i haskellPackages.postgrest
 
-If you use Windows, you can install PostgREST using `Chocolatey <https://chocolatey.org>`_ or `Scoop <https://scoop.sh>`_.
+If you use Windows, you can install PostgREST using `Chocolatey <https://chocolatey.org/packages/postgrest>`_ or `Scoop <https://scoop.sh>`_.
 
 .. code:: bash
 
   choco install postgrest
   scoop install postgrest
 
-When a pre-built binary does not exist for your system you can :ref:`build the project from source <build_source>`.
+.. _build_source:
+
+Building from Source
+--------------------
+
+When a pre-built binary does not exist for your system you can build the project from source.
+
+.. note::
+
+  We discourage building and using PostgREST on **Alpine Linux** because of a reported GHC memory leak on that platform.
+
+You can build PostgREST from source with `Stack <https://github.com/commercialhaskell/stack>`_. It will install any necessary Haskell dependencies on your system.
+
+* `Install Stack <http://docs.haskellstack.org/en/stable/README.html#how-to-install>`_ for your platform
+* Install Library Dependencies
+
+  =====================  =======================================
+  Operating System       Dependencies
+  =====================  =======================================
+  Ubuntu/Debian          libpq-dev, libgmp-dev, zlib1g-dev
+  CentOS/Fedora/Red Hat  postgresql-devel, zlib-devel, gmp-devel
+  BSD                    postgresql95-client
+  OS X                   libpq, gmp
+  =====================  =======================================
+
+* Build and install binary
+
+  .. code-block:: bash
+
+    git clone https://github.com/PostgREST/postgrest.git
+    cd postgrest
+
+    # adjust local-bin-path to taste
+    stack build --install-ghc --copy-bins --local-bin-path /usr/local/bin
+
+.. note::
+
+   - If building fails and your system has less than 1GB of memory, try adding a swap file.
+   - `--install-ghc` flag is only needed for the first build and can be omitted in the subsequent builds.
 
-Running
--------
+* Check that the server is installed: :code:`postgrest --help`.
+
+To properly run the test suite, you need a PostgreSQL database that the tests can run against. There are several ways to set up this database.
+
+Testing with a temporary database
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I think this part should be removed as well. It seems off from the "Installation" page, which should be sort of a quick read.

Additionally, I haven't been using stack for a long time now, only nix :-]